odoo/addons/base/models/ir_qweb_fields.py

```diff
@@ -660,6 +660,9 @@ def value_to_html(self, value, options):
         if not value:
             return ''
 
+        if not isinstance(value, models.Model):
+            raise ValueError(_("The value send to contact field is not a valid model."))
```

Comment:
This error makes no sense. Nobody sent anything to anyone, and "contact" is not a field, it's a widget.

The error message should be something along the lines of "the contact widget is only allowed on a partner field". And it should be a TypeError.

Also if we're checking this, we should proably also check that the model is a `res.partner`.